### PR TITLE
MCOL-3607 Add long double support to math functions

### PR DESCRIPTION
**Pull request by davidhall586**

### CODE REVIEW

File: utils/funcexp/func_math.cpp [hunk 1]
```diff
@@ -348,6 +378,32 @@ double Func_atan::getDoubleVal(Row& row,
         }
         break;
 
+        case execplan::CalpontSystemCatalog::LONGDOUBLE:
+        {
+            // null value is indicated by isNull
+            long double value = parm[0]->data()->getLongDoubleVal(row, isNull);
+
+            if (isNull || (value < -1.0 || value > 1.0))
```
Comment (jmrojas2332) [inline utils/funcexp/func_math.cpp]: does this conditional contain a typo, perhaps? should it not match line 358 above?

Comment (jmrojas2332) [reply utils/funcexp/func_math.cpp]: Val can be less than -1 and greater than 1 when taking the atan(val) if i'm not mistaken

File: utils/funcexp/func_math.cpp [hunk 2]
```diff
@@ -1316,6 +1514,21 @@ double Func_tan::getDoubleVal(Row& row,
         }
         break;
 
+        case execplan::CalpontSystemCatalog::LONGDOUBLE:
+        {
+            // null value is indicated by NaN
+            long double value = parm[0]->data()->getLongDoubleVal(row, isNull);
+
+            if (isNull || value <= 0.0)
```
Comment (jmrojas2332) [inline utils/funcexp/func_math.cpp]: This may be a typo. I believe it should match line 1507. We could take the tangent of a negative

**Review by jmrojas2332 — COMMENTED** [review]
Made a couple of comments on this regarding possibly a couple of typos.

File: utils/funcexp/func_math.cpp [hunk 3]
```diff
@@ -491,6 +547,21 @@ double Func_cos::getDoubleVal(Row& row,
         }
         break;
 
+        case execplan::CalpontSystemCatalog::LONGDOUBLE:
+        {
+            // null value is indicated by isNull
+            long double value = parm[0]->data()->getLongDoubleVal(row, isNull);
+
+            if (isNull || (value < -1.0 || value > 1.0))
```
Comment (jmrojas2332) [inline utils/funcexp/func_math.cpp]: I believe this conditional should match line 540. Domain of cosine can be less than 1 and greater than 1

File: utils/funcexp/func_math.cpp (hunk 1 above)
Comment (LinuxJedi) [reply utils/funcexp/func_math.cpp]: −π/2 and +π/2 radians are the limits.

Comment (davidhall586) [reply utils/funcexp/func_math.cpp]: You are correct. Typo

File: utils/funcexp/func_math.cpp (hunk 3 above)
Comment (davidhall586) [reply utils/funcexp/func_math.cpp]: Correct.

File: utils/funcexp/func_math.cpp (hunk 2 above)
Comment (davidhall586) [reply utils/funcexp/func_math.cpp]: Also correct

**Review by davidhall586 — COMMENTED** [review]
Corrected typos

**Review by jmrojas2332 — APPROVED** [review]
Looks good.

**Pull request merged**